### ISSUE TITLE
Nullability mismatch not reported for event assignment

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31018]
The following should result in a warning for `c.E += y`:
```c#
#pragma warning disable 0067
using System;
class C
{
    event Action<object?> E;
    static void F(Action<object?> x, Action<object> y)
    {
        var c = new C();
        c.E += x;
        c.E += y; // warning
    }
}
```

**Comment by jaredpar** [comment]
Related to #29839
